pkg/controllers/managementuser/rbac/podsecuritypolicy/binding.go

```diff
@@ -90,6 +95,11 @@ func (l *lifecycle) sync(obj *v3.PodSecurityPolicyTemplateProjectBinding) (runti
 		return obj, nil
 	}
 
+	if err := checkClusterVersion(l.clusterName, l.clusters); err != nil && !goerrors.Is(err, errVersionIncompatible) {
```

Comment:
If the version check in deferred.go prevents these handlers from being registered, are these in-handler checks necessary?

Reply:
Yes, because registration runs only once - when you provision a remote cluster with PSPs enabled, or when you enable them afterwards in cluster configuration.
That means you could have a remote v1.24 cluster with PSP enabled (so the controllers will be registered and started). When you upgrade the remote cluster to v1.25 (with PSPs disabled), the controllers will not de-register themselves or stop. We don't want them to run, since PSPs are irrelevant on v1.25. Exiting from `sync` early is, unfortunately, the only way, unless there is a clean way to de-register controllers.